src/librustc_resolve/late/diagnostics.rs

```diff
@@ -1301,26 +1360,26 @@ impl<'tcx> LifetimeContext<'_, 'tcx> {
 
         match (lifetime_names.len(), lifetime_names.iter().next(), snippet.as_deref()) {
```

Comment:
this match is somewhat hard to read, considering that perf is somewhat irrelevant here, I think it might be a good idea to convert `lifetime_names` to a `Vec` and use slice patterns here

Reply:
I think this code predates slice patterns. Good idea.

Reply:
done